torch/_dynamo/package.py

```diff
@@ -184,12 +210,22 @@ def _initialize(self, fn: Any, dynamo: Optional[_DynamoCacheEntry] = None) -> No
                 raise RuntimeError(
                     f"Compile package was created with a different PyTorch version: {dynamo.torch_version}"
                 )
+            for code in dynamo.inlined_sources:
+                m = importlib.import_module(code.module)
+                checksum = _hash_sourcelines(m, code.firstlineno, code.lastlineno)
+                if checksum != code.checksum:
```

Comment:
I don't really get why we're RuntimeErroring; shouldn't we just use the compiler stance? If the compiler stance is not error on recompile, we should just ignore this package and recompile.

Reply:
If we treat this as cache, then it makes sense to ignore this. In an API where we call load() directly on a single model, imo we should always throw an error because load() shouldn't silently be a noop.

For load_cache_artifacts(), I don't have strong opinions whether to error or not (depending on how strong the guarantee is required to be).

I agree we should have a config somewhere governing the behavior here, not sure stance is a good option here because technically loading is not part of compilation today.